src/bittybuzz/src/BittyBuzzClosureRegister.cpp

```diff
@@ -0,0 +1,43 @@
+#include "BittyBuzzClosureRegister.h"
+#include <bbzvm.h>
+#include <cstring>
+
+BittyBuzzClosureRegister::BittyBuzzClosureRegister() = default;
+
+bool BittyBuzzClosureRegister::registerClosure(const char* functionName,
+                                               bbzheap_idx_t closureHeapIdx) {
+    if (m_closureRegistersLength >= m_maxSize) {
```

Comment:
What happens in the case that the m_closureRegisterLength has exceeded its maxlength? Do those values directly correspond to the size of the heap used? Is it corruputed or is there some protections internally to the bbvm?

Similar to the previos comment on the interface of this class, the allocation could be done internally to avoid this sort of weird state.

Reply:
I'm not sure what you mean, here it's really just a list/array of addresses in the heap associated with a string. It shouldn't exceed since there is a check at this line. the m_maxSize is the max size of the array. There is no allocation here, It's already allocated by the buzz script, I just make it permanent on line 13 so the GC doesn't destroy it (so the closure lives during the whole program)